tests/components/iaqualink/test_init.py

```diff
@@ -0,0 +1,360 @@
+
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+    assert DOMAIN not in hass.data
+
+
+async def test_multiple_updates(hass, config_entry):
+    """Test all possible results of online status transition after update."""
+    config_entry.add_to_hass(hass)
+
+    client = get_aqualink_client()
+    system = get_aqualink_system(client)
+    systems = {system.serial: system}
+
+    with patch(
+        "homeassistant.components.iaqualink.AqualinkClient.login", return_value=None
+    ), patch(
+        "homeassistant.components.iaqualink.AqualinkClient.get_systems",
+        return_value=systems,
+    ), patch(
+        "iaqualink.system.AqualinkSystem.get_devices", return_value={}
+    ):
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert config_entry.state == ConfigEntryState.LOADED
+
+    def set_online_to_true():
+        system.online = True
+
+    def set_online_to_false():
+        system.online = False
+
+    # True -> True
+    system.online = True
+    with patch(
+        "iaqualink.system.AqualinkSystem.update", side_effect=set_online_to_true
+    ), patch("homeassistant.components.iaqualink._LOGGER.warning") as mock_warn:
+        _ffwd_next_update_interval(hass)
+        await hass.async_block_till_done()
+        mock_warn.assert_not_called()
+
+    # True -> False
+    system.online = True
+    with patch(
+        "iaqualink.system.AqualinkSystem.update", side_effect=set_online_to_false
+    ), patch("homeassistant.components.iaqualink._LOGGER.warning") as mock_warn:
+        _ffwd_next_update_interval(hass)
+        await hass.async_block_till_done()
+        mock_warn.assert_not_called()
+
+    # True -> None / ServiceException
+    system.online = True
+    with patch(
+        "iaqualink.system.AqualinkSystem.update",
+        side_effect=AqualinkServiceException,
+    ), patch("homeassistant.components.iaqualink._LOGGER.warning") as mock_warn:
+        _ffwd_next_update_interval(hass)
+        await hass.async_block_till_done()
+        mock_warn.assert_called_once()
+
+    # False -> False
+    system.online = False
+    with patch(
+        "iaqualink.system.AqualinkSystem.update", side_effect=set_online_to_false
+    ), patch("homeassistant.components.iaqualink._LOGGER.warning") as mock_warn:
+        _ffwd_next_update_interval(hass)
+        await hass.async_block_till_done()
+        mock_warn.assert_not_called()
+
+    # False -> True
+    system.online = False
+    with patch(
+        "iaqualink.system.AqualinkSystem.update", side_effect=set_online_to_true
+    ), patch("homeassistant.components.iaqualink._LOGGER.warning") as mock_warn:
+        _ffwd_next_update_interval(hass)
+        await hass.async_block_till_done()
+        mock_warn.assert_called_once()
+
+    # False -> None / ServiceException
+    system.online = False
+    with patch(
+        "iaqualink.system.AqualinkSystem.update",
```

Comment:
Patch the library where its imported instead of directly

Reply:
example https://github.com/home-assistant/core/pull/53745/files#diff-5293d84f85a25723ed5e2661903b1cd47231805b0f92e1cd8b3ca6e82e6dc972R52

Reply:
It's not imported in home-assistant. Are you suggesting I patch iaqualink.client.AqualinkSystem instead of iaqualink.system.AqualinkSystem (since client imports system.AqualinkSystem). I'm not sure I see the benefit?

Reply:
Patch `AqualinkSystem` where its used in the integration and change the `update` mock here instead

Reply:
You may need to patch `homeassistant.components.iaqualink.AqualinkClient.get_systems` to return a autospeced mocked object